src/ui/services/networkService.ts

```diff
@@ -23,50 +23,72 @@ export class NetworkService extends EventEmitter {
             this.currentConnection.on("connect", resolve);
-
-            const [add, update, remove] = this.buildListUpdater("lobbylist", this.lobbies, {
+            var userInfo = {
+                username: username,
+                iconId: iconId
+            };
+            this.me = {id: -1, username: username, iconId: iconId};
+            this.currentConnection.emit("user.userInfo", userInfo);
+            this.currentConnection.on("user.userInfo", (c) => {
+                this.me = c;
+            });
+            
+            const [addLobby, updateLobby, removeLobby] = this.buildListUpdater("lobbylist", this.lobbies, {
+                id: "id",
                 name: "name",
-                creator: "creator",
+                owner: "owner",
                 playerLimit: "playerLimit",
                 playerCount: "playerCount",
-                gamemodeName: "gameMode",
+                gamemodeName: "gamemodeName",
                 hasPassword: "requirePassword",
                 address: "address",
                 port: "port"
             });
-            this.currentConnection.on("lobbylist-add", add);
-            this.currentConnection.on("lobbylist-update", update);
-            this.currentConnection.on("lobbylist-remove", remove);
+            this.currentConnection.on("lobbylist-add", addLobby);
+            this.currentConnection.on("lobbylist-update", updateLobby);
+            this.currentConnection.on("lobbylist-remove", removeLobby);
+
+            const [addUser, updateUser, removeUser] = this.buildListUpdater("users", this.users, {
+                id: "id",
+                username: "username",
+                idIcon: "idIcon"
+            });
+            this.currentConnection.on("users-add", addUser);
+            this.currentConnection.on("users-update", updateUser);
+            this.currentConnection.on("users-remove", removeUser);
 
             //Sending the options to the server
-            var options = {
-                name: "name",
-                creator: "creator",
-                playerLimit: "10",
-                gamemodeName: "gameMode",
-                requirePassword: "false",
-                address: "address",
-                port: "port"
-            }
-            this.currentConnection.emit('lobby.create', options);
+            this.currentConnection.on("lobby.list", function(lobbiesList){
+                for(var i = 0; i < lobbiesList.length; i++){
+                    addLobby(lobbiesList[i]);
+                }
+            });
             this.currentConnection.emit('lobby.list');
-            this.currentConnection.on("lobby.list", function(lobbies){
-                var count = 0;
-                while (count <= lobbies.length){
-                    add(lobbies[count]);
-                    count++;
+            
+            this.currentConnection.on('user.list', function(clientsList){
```

Comment:
Arrow function, `for (const lobby of clientsList)`

Reply:
d